### ISSUE TITLE
GLMAT: Numerical instability

### ISSUE THREAD

**Issue by sbenkorichi** [firemodels/fds#5669]
Marcos,
Looks like you're gonna be busy for a while. Taking the input file reported by the user. I can't run the whole meshes, so, I set it to 6 Meshes. I get numerical instability after few seconds.
```
 Job TITLE        : 
 Job ID string    : 1

 Time Step:      1, Simulation Time:      0.14 s
 Time Step:      2, Simulation Time:      0.14 s
 Time Step:      3, Simulation Time:      0.14 s
 Time Step:      4, Simulation Time:      0.14 s
 Time Step:      5, Simulation Time:      0.14 s
 Time Step:      6, Simulation Time:      0.14 s
MPI process      0 has completed

STOP: Numerical Instability (CHID: 1)

```
[GLMAT.fds.txt](https://github.com/firemodels/fds/files/1440982/GLMAT.fds.txt)

**Comment by rmcdermo** [comment]
@sbenkorichi I assume you've tried this case without GLMAT and it works?

**Comment by sbenkorichi** [comment]
Yes, It appears only when the GLMAT is enabled. I'm trying to see if I can corner this NI.

**Comment by sbenkorichi** [comment]
Here is another similar example where It ran successfully till the end. I'm not sure yet the reason behind, either the Cell numbers, or how the meshes are configured. 
[test.fds.txt](https://github.com/firemodels/fds/files/1441092/test.fds.txt)

**Comment by marcosvanella** [comment]
I'll take a look.

**Comment by marcosvanella** [comment]
Salah, I see the input file has 32 meshes.
Can you post you failing 6 mesh case you have and the whole info written to screen you see?
Thank you.

**Comment by marcosvanella** [comment]
Sorry, I just see you commented the rest.

**Comment by sbenkorichi** [comment]
Marcos,
I will give you more feedback later on this when I get to it.

**Comment by marcosvanella** [comment]
Salah, are you running with the bundle or gnu compiled code?
I can't reproduce your problem.

**Comment by sbenkorichi** [comment]
Yes, gnu compiler. 
Sorry, I forget to mention that. 
That's when it's run in serial: I get such issue. 
In parallel: 
```
$ mpirun -np 6 /home/salah/firemodels/fds/Build/mpi_gnu_linux_64/fds_mpi_gnu_linux_64 GLMAT.fds
--------------------------------------------------------------------------
There are not enough slots available in the system to satisfy the 6 slots
that were requested by the application:
  /home/salah/firemodels/fds/Build/mpi_gnu_linux_64/fds_mpi_gnu_linux_64

Either request fewer slots for your application, or make more slots available
for use.
--------------------------------------------------------------------------
```

**On Intel:**
Parallel: Works.

Serial: Doesn't work as Gnu:

**Comment by marcosvanella** [comment]
Salah, for the gnu (openmpi 3.0) parallel submission add the `--oversubscribe` flag after mpirun.

**Comment by sbenkorichi** [comment]
Marcos,
```
$ mpirun --oversubscribe  -np 6 /home/salah/firemodels/fds/Build/mpi_gnu_linux_64/fds_mpi_gnu_linux_64 GLMAT.fds
[warn] Epoll MOD(1) on fd 14 failed.  Old events were 6; read change was 0 (none); write change was 2 (del): Bad file descriptor
[warn] Epoll MOD(4) on fd 14 failed.  Old events were 6; read change was 2 (del); write change was 0 (none): Bad file descriptor
[warn] Epoll MOD(1) on fd 14 failed.  Old events were 6; read change was 0 (none); write change was 2 (del): Bad file descriptor
[warn] Epoll MOD(4) on fd 14 failed.  Old events were 6; read change was 2 (del); write change was 0 (none): Bad file descriptor
[warn] Epoll MOD(1) on fd 14 failed.  Old events were 6; read change was 0 (none); write change was 2 (del): Bad file descriptor
[warn] Epoll MOD(4) on fd 14 failed.  Old events were 6; read change was 2 (del); write change was 0 (none): Bad file descriptor
[warn] Epoll MOD(1) on fd 14 failed.  Old events were 6; read change was 0 (none); write change was 2 (del): Bad file descriptor
[warn] Epoll MOD(4) on fd 14 failed.  Old events were 6; read change was 2 (del); write change was 0 (none): Bad file descriptor
[warn] Epoll MOD(1) on fd 14 failed.  Old events were 6; read change was 0 (none); write change was 2 (del): Bad file descriptor
[warn] Epoll MOD(4) on fd 14 failed.  Old events were 6; read change was 2 (del); write change was 0 (none): Bad file descriptor
[warn] Epoll MOD(1) on fd 14 failed.  Old events were 6; read change was 0 (none); write change was 2 (del): Bad file descriptor
[warn] Epoll MOD(4) on fd 14 failed.  Old events were 6; read change was 2 (del); write change was 0 (none): Bad file descriptor
```
Keeps hanging.

**Comment by marcosvanella** [comment]
Do you see this last issue commenting the &PRES line?

**Comment by sbenkorichi** [comment]
Yes,
First, let's fix the issue with serial then we move to parallel: 
Without &PRES line

```
$ fds-gnu GLMAT.fds
 Mesh 1 is assigned to MPI Process 0
 Mesh 2 is assigned to MPI Process 0
 Mesh 3 is assigned to MPI Process 0
 Mesh 4 is assigned to MPI Process 0
 Mesh 5 is assigned to MPI Process 0
 Mesh 6 is assigned to MPI Process 0
 Completed Initialization Step  1
 Completed Initialization Step  2
 Completed Initialization Step  3
 Completed Initialization Step  4

 Fire Dynamics Simulator

 Current Date     : November  3, 2017  16:06:53
 Version          : FDS 6.6.0
 Revision         : FDS6.6.0-131-g88ae75a-dirty-master
 Revision Date    : Wed Nov 1 16:03:29 2017 -0400
 Compiler         : Gnu gfortran 7.2.0-1ubuntu1~16.04)
 Compilation Date : Nov 03, 2017  14:39:53

 MPI Enabled;    Number of MPI Processes:       1
 OpenMP Enabled; Number of OpenMP Threads:      1

 MPI version: 3.1
 MPI library version: Open MPI v3.0.0, package: Open MPI salah@sbenkorichi Distribution, ident: 3.0.0, repo rev: v3.0.0, Sep 12, 2017

 Job TITLE        : 
 Job ID string    : glmat

 Time Step:      1, Simulation Time:      0.14 s
 Time Step:      2, Simulation Time:      0.27 s
 Time Step:      3, Simulation Time:      0.41 s
 Time Step:      4, Simulation Time:      0.54 s
 Time Step:      5, Simulation Time:      0.68 s
 Time Step:      6, Simulation Time:      0.81 s
 Time Step:      7, Simulation Time:      0.95 s
 Time Step:      8, Simulation Time:      1.09 s
 Time Step:      9, Simulation Time:      1.22 s
 Time Step:     10, Simulation Time:      1.36 s
 Time Step:     20, Simulation Time:      2.71 s

```

With &PRES line
```
$ fds-gnu GLMAT.fds
 Mesh 1 is assigned to MPI Process 0
 Mesh 2 is assigned to MPI Process 0
 Mesh 3 is assigned to MPI Process 0
 Mesh 4 is assigned to MPI Process 0
 Mesh 5 is assigned to MPI Process 0
 Mesh 6 is assigned to MPI Process 0
 Completed Initialization Step  1
 Completed Initialization Step  2
 Completed Initialization Step  3
 Completed Initialization Step  4
  
 Using GLMAT as pressure solver. List of H unknown numbers per proc:
 MYID=       0, NUNKH_LOCAL=  305168

 Fire Dynamics Simulator

 Current Date     : November  3, 2017  16:08:03
 Version          : FDS 6.6.0
 Revision         : FDS6.6.0-131-g88ae75a-dirty-master
 Revision Date    : Wed Nov 1 16:03:29 2017 -0400
 Compiler         : Gnu gfortran 7.2.0-1ubuntu1~16.04)
 Compilation Date : Nov 03, 2017  14:39:53

 MPI Enabled;    Number of MPI Processes:       1
 OpenMP Enabled; Number of OpenMP Threads:      1

 MPI version: 3.1
 MPI library version: Open MPI v3.0.0, package: Open MPI salah@sbenkorichi Distribution, ident: 3.0.0, repo rev: v3.0.0, Sep 12, 2017

 Job TITLE        : 
 Job ID string    : glmat

 Time Step:      1, Simulation Time:      0.14 s
 Time Step:      2, Simulation Time:      0.14 s
 Time Step:      3, Simulation Time:      0.14 s
 Time Step:      4, Simulation Time:      0.14 s
 Time Step:      5, Simulation Time:      0.14 s
 Time Step:      6, Simulation Time:      0.14 s
MPI process      0 has completed

STOP: Numerical Instability (CHID: glmat)
```

This is the file that I used: 
```
&HEAD CHID='glmat'/
&TIME T_END=10.0,  /
 
&MESH ID='exit1-merged-a', IJK=29,70,50, XB=2332.0,2346.5,-1.0,13.0,0.0,10.0,                /
&MESH ID='exit3', IJK=40,70,21, XB=2350.0,2370.0,-1.0,13.0,2.6,6.8,                          /
&MESH ID='05 02 02-a-b-b-b-b-b-b-b-b-b-b-b', IJK=24,48,34, XB=2304.5,2316.5,0.0,9.6,0.0,6.8, /
&MESH ID='exit1-merged-b', IJK=7,70,50, XB=2346.5,2350.0,-1.0,13.0,0.0,10.0,                 /
&MESH ID='exit2', IJK=40,70,13, XB=2350.0,2370.0,-1.0,13.0,0.0,2.6,                          /
&MESH ID='exit4', IJK=40,70,16, XB=2350.0,2370.0,-1.0,13.0,6.8,10.0,                        /


&PRES SOLVER='GLMAT'/
 
&TAIL /
```

**Comment by sbenkorichi** [comment]
This is not an issue. 
Sorry Marcos for giving you troubles. 
Specifying &PRES on separate meshes caused this issue. I should have looked into this more in detail and examine it with smokeview. 
I corrected the meshes linkage and no errors received.
